Node.js/samplesubscribe.js

```diff
@@ -60,16 +60,30 @@ mqBoilerPlate.initialise('SUBSCRIBE')
     debug_info('Getting Messages');
     return mqBoilerPlate.getMessages(null, msgCB);
   })
+  .then(() => {
+    debug_info('Kick start the get callback');
+    return mqBoilerPlate.startGetAsyncProcess();
+  })  
   .then(() => {
     debug_info('Waiting for termination');
     return mqBoilerPlate.checkForTermination();
   })
+  .then(() => {
+    debug_info('Signal termination of the callback thread');
+    return mqBoilerPlate.signalDone();
+  })    
   .then(() => {
     mqBoilerPlate.teardown();
   })
+  .then(() => {
+    debug_info("Application Completed");
+    process.exit(0);
+  })
   .catch((err) => {
+    debug_warn(err);
     mqBoilerPlate.teardown();
   })
-
-
-debug_info('Application Completed');
+  .then(() => {
```

Comment:
This final then won't be called. If there is an error, then this .then needs to be on the teardown in the catch. ie.

mqBoilerPlate.teardown().then(() => process.exit(1))